### PR TITLE
Modernize Python 2 code to get ready for Python 3

### PR DESCRIPTION
**Pull request by cclauss**
Make the minimal, safe changes required to convert the repo's code to be syntax compatible with both Python 2 and Python 3.  There may be other changes required to complete a port to Python 3 but this PR is a minimal, safe first step.

Run: [futurize --stage1 -w **/*.py](http://python-future.org/futurize_cheatsheet.html)

See Stage 1: "safe" fixes http://python-future.org/automatic_conversion.html#stage-1-safe-fixes

### CODE REVIEW

File: test/checks/fuerte/distro-to-install.py [hunk 1]
```diff
@@ -1,16 +1,17 @@
 #!/usr/bin/env python
 
+from __future__ import print_function
```
Comment (dirk-thomas) [inline test/checks/fuerte/distro-to-install.py]: This is not necessary in this case (see http://wiki.ros.org/python_2_and_3_compatible_code#Print_is_now_a_function).

Same below for many other cases.

File: test/local_tests/test_with_mock_workspace.py [hunk 2]
```diff
@@ -74,7 +75,7 @@ def test_noproject(self):
 
         out = self.cmake(CMAKE_PREFIX_PATH=self.installdir,
                          expect=fail)
-        print("failed as expected, out=", out)
+        print(("failed as expected, out=", out))
```
Comment (dirk-thomas) [inline test/local_tests/test_with_mock_workspace.py]: The extra parenthesis don't seem to make any sense. Why are they being introduced?

**Review by dirk-thomas — CHANGES_REQUESTED** [review]

**Comment by cclauss** [comment]
The import keeps Python 2 developers from breaking Python 3 code because when the import is used then print without the parens is a syntax error in both Python 2 and Python 3.  The import also deals gracefully with commas inside a print statement thus the double parens for trying to preserve the output...

$ python2
```
Python 2.7.14 ....
>>> print 0, 1
0 1
>>> print(0, 1)
(0, 1)
>>> print((0, 1))
(0, 1)
>>> from __future__ import print_function
>>> print 0, 1
  File "<stdin>", line 1
    print 0, 1
          ^
SyntaxError: invalid syntax
>>> print(0, 1)
0 1
>>> print((0, 1))
(0, 1)
```

File: test/network_tests/test_basics.py [hunk 3]
```diff
@@ -1,11 +1,12 @@
 #!/usr/bin/env python
 
+from __future__ import absolute_import
```
Comment (dirk-thomas) [inline test/network_tests/test_basics.py]: Why is this needed here?

Same question below.

**Comment by dirk-thomas** [comment]
> The import keeps Python 2 developers from breaking Python 3 code because when the import is used then print without the parens is a syntax error in both Python 2 and Python 3. The import also deals gracefully with commas inside a print statement thus the double parens for trying to preserve the output...

I understand. The wiki page I referenced mentions that. I just don't think it should be used when not necessary.

Wasn't there one occasion where print is called with multiple arguments? In that the future import would make sense. But I can't see it anywhere anymore since you force pushed.

File: .travis.yml [hunk 4]
```diff
@@ -4,7 +4,12 @@ python:
   - "3.4_with_system_site_packages"
 # command to install dependencies
 install:
-  - pip install argparse catkin-pkg empy mock nose
+  - pip install argparse catkin-pkg empy flake8 mock nose
+before_script:
+  # stop the build if there are Python syntax errors or undefined names
+  - flake8 . --count --select=E9,F821,F822,F823 --show-source --statistics
+  # exit-zero treats all errors as warnings.  The GitHub editor is 127 chars wide
+  - flake8 . --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics
```
Comment (dirk-thomas) [inline .travis.yml]: Please move the check into a unit test so that it is being run locally too.

**Comment by cclauss** [comment]
> Wasn't there one occasion where print is called with multiple arguments?
https://github.com/ros/catkin/blob/kinetic-devel/test/local_tests/test_with_mock_workspace.py#L77

File: test/network_tests/test_basics.py (hunk 3 above)
Comment (cclauss) [reply test/network_tests/test_basics.py]: https://www.python.org/dev/peps/pep-0328/#rationale-for-absolute-imports

**Comment by dirk-thomas** [comment]
> https://github.com/ros/catkin/blob/kinetic-devel/test/local_tests/test_with_mock_workspace.py#L77

For that line future import would make sense. I doubt the intention in Python 2 was to print parenthesis around the arguments.

**Comment by cclauss** [comment]
I used __str.format()__ instead of the print_function import to make the intention explicit.

**Review by dirk-thomas — APPROVED** [review]

**Comment by dirk-thomas** [comment]
Looks good to me now. Thank you for creating the patch and iterating on it.
